Porting the build system to cmake

ikrukov opened the pull request.

geometer commented:
Found two regressions against the original version.
1) It does not use ccache, which increases compile time during development.
2) It does not build KnotEditor.app bundle on Mac.

ikrukov commented:
> 1. It does not use ccache. That increases compile time during development.

Fixed

> 2\. It does not build KnotEditor.app bundle on Mac.

Fixed

geometer commented:
Some files are missing from the bundle, particularly the Contents/Resources folder containing icon files. I compare with the dmg image in the "pre-built" folder. I'm not sure all the files in the dmg are necessary. Icons are.
